OpenRA.Mods.Common/Lint/CheckSequences.cs

```diff
@@ -22,12 +22,36 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Map map)
 			var sequences = MiniYaml.MergeLiberal(map.SequenceDefinitions, Game.ModData.Manifest.Sequences.Select(s => MiniYaml.FromFile(s)).Aggregate(MiniYaml.MergeLiberal));
 
 			foreach (var actorInfo in map.Rules.Actors)
+			{
 				foreach (var renderInfo in actorInfo.Value.Traits.WithInterface<RenderSimpleInfo>())
 				{
 					var image = renderInfo.Image ?? actorInfo.Value.Name;
 					if (!sequences.Any(s => s.Key == image.ToLowerInvariant()) && !actorInfo.Value.Name.Contains("^"))
 						emitWarning("Sprite image {0} from actor {1} has no sequence definition.".F(image, actorInfo.Value.Name));
 				}
+
+				foreach (var traitInfo in actorInfo.Value.Traits.WithInterface<ITraitInfo>())
+				{
+					foreach (var field in traitInfo.GetType().GetFields())
+					{
+						if (field.HasAttribute<SequenceReferenceAttribute>())
+						{
+							var sequence = (string)field.GetValue(traitInfo);
+							if (string.IsNullOrEmpty(sequence))
+								continue;
+
+							var renderInfo = actorInfo.Value.Traits.WithInterface<RenderSimpleInfo>().FirstOrDefault();
```

Comment:
This needs to be using `RenderSpritesInfo.GetImage` if you want it to work in all cases.